ols/src/llms/llm_loader.py

```diff
@@ -43,7 +43,7 @@ def _resolve_provider_config(
 
     provider_config = providers_config.providers.get(provider)
 
-    if model not in provider_config.models:
+    if provider != constants.PROVIDER_FAKE and model not in provider_config.models:
```

Comment:
why do we need to special case this?  Can we not just provide a full fake provider/model configuration?

or at least during config loading, if we see the requested provider is the fake one, populate the rest of the configuration as if it was a normal provider?  (using appropriately fake values, of course)?  Which sort of seems to be what you're trying to do with the validation logic you added, but i don't think such logic belongs in "validation"

Reply:
Removed this special case according to the reason stated above.